src/txn/txn_rollback_to_stable.c

```diff
@@ -371,7 +371,7 @@ __rollback_ondisk_fixup_key(WT_SESSION_IMPL *session, WT_REF *ref, WT_ROW *rip,
      * outside the constraints of transactions. Therefore, there is no need for snapshot based
      * visibility checks.
      */
-    F_SET(hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+    F_SET(hs_cursor, WT_CURSTD_HS_READ_ALL);
```

Comment:
I assume it's fine to have a value from history store that might have globally visible tombstone?

Reply:
Yes. All the global visible tombstones according to the recovered checkpoint snapshot are removed by RTS. Whereas the global visibility check in the HS is carried out based on the current global visibility. Anyway, it is protected now by avoiding the increase of the oldest ID. I feel it is better to handle the visibility in the RTS than in the HS.